lib/utils/bcrypt.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"golang.org/x/crypto/bcrypt"
+)
+
+const maxInputSize = 72
+
+// truncateToMaxSize Make sure input is truncated to the maximum length crypto accepts.  Crypto changed the behavior
+// from ignoring the extra input to returning an error, this truncation is necessary to maintain compatibility with
+// customers who have long passwords, or more commonly our recovery codes.
+func truncateToMaxSize(input []byte) []byte {
+	if len(input) > maxInputSize {
+		return input[:maxInputSize]
+	}
```

Comment:
I don't think this is the correct way of dealing with this issue. I agree that we should truncate passwords that are already in our DB, but I also think we should return the error when someone tries to set a password that is longer than 72 characters. Otherwise, we give people a false sense of security.
Another question is, do we really want to limit the password to 72 chars? In my opinion, sounds reasonable, tbh

I also think that we should mention in our documentation what passwords we support. 
WDYT?

Reply:
> I agree that we should truncate passwords that are already in our DB

At first I expected to need to do this for verifying passwords also, however the cryto implementation does still silently truncate when verifying.  So I don't believe any changes are necessary for supporting existing existing passwords and recovery tokens.

> I also think we should return the error when someone tries to set a password that is longer than 72 characters

I was on the fence about this.  You could argue that by continuing to accept truncated passwords we are perpetuating a false sense of security either way (but this is necessary to support existing passwords and reset tokens).  I felt like this implementation is the most consistent, but I agree with the reasons crypto made the changes to start returning an error on larger inputs (I just wish they had started that way).

That said, it's also worth recognizing that 72 characters is more than enough entropy to prevent any realistic brute force attack.  Although more characters is theoretically better, there is certainly an argument that it's also unnecessary.

If there is a preference to instead prevent passwords and recovery tokens > 72 characters I am happy to make that change.

Reply:
@r0mant @zmb3 Any thoughts? Do we want to:
1. Block people from using passwords longer than 72 chars by returning an error if someone tries to do it?
2. Keep truncating as we used to.